pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java

```diff
@@ -55,8 +56,10 @@ public SegmentFetcherAndLoader(@Nonnull Configuration config, @Nonnull InstanceD
     Configuration pinotFSConfig = config.subset(CommonConstants.Controller.PREFIX_OF_CONFIG_OF_PINOT_FS_FACTORY);
     Configuration segmentFetcherFactoryConfig =
         config.subset(CommonConstants.Server.PREFIX_OF_CONFIG_OF_SEGMENT_FETCHER_FACTORY);
+    Configuration pinotCrypterConfig = config.subset(CommonConstants.Controller.PREFIX_OF_CONFIG_OF_PINOT_CRYPTER_FACTORY);
 
     PinotFSFactory.init(pinotFSConfig);
+    PinotCrypterFactory.init(pinotCrypterConfig);
```

Comment:
Should these factories (FS and Crypter) be initialized in the Controller/Server starters? Why is it being initialized here?

Reply:
This is where we initialize pinotfsfactory, segment fetcher, it is called in server starter.
unit tests were added in the last pr for pinot crypter factory initialization